modules/camera/camera_picamera2.py

```diff
@@ -11,11 +11,54 @@
     pass
 
 from . import base_camera
-from . import camera_configurations
 
 
-# TODO: pass in as constructor parameter
-CAMERA_TIMEOUT = 1
+class ConfigPiCamera2:
+    """
+    Configuration for the PiCamera.
+    """
+
+    def __init__(
+        self,
+        timeout: float = 1.0,
+        exposure_time: int = 250,
+        analogue_gain: float = 64.0,
+        contrast: float = 1.0,
+        maybe_lens_position: float | None = None,
+    ) -> None:
+        """
+        timeout: Getting image timeout in seconds.
+
+        exposure_time: Microseconds.
+        analogue_gain:
+        contrast: 0.0 to 32.0 . 0.0 is no contrast, 1.0 is normal contrast, higher is more contrast.
+        lens_position: Position of the lens is dioptres (reciprocal of metres: 1/m ).
+        """
+        self.timeout = timeout
+
+        self.exposure_time = exposure_time
+        self.analogue_gain = analogue_gain
+        self.contrast = contrast
+        self.maybe_lens_position = maybe_lens_position
+
+    def to_dict(self) -> dict[str, int | float]:
+        """
+        Dictionary containing camera controls.
+        """
+        camera_controls = {
+            "ExposureTime": self.exposure_time,
+            "AnalogueGain": self.analogue_gain,
+            "Contrast": self.contrast,
+        }
+
+        if self.maybe_lens_position is not None:
+            camera_controls["LensPosition"] = self.maybe_lens_position
+            camera_controls["AfMode"] = picamera2.controls.AfModeEnum.Manual
```

Comment:
I think this needs an import: `from libcamera import controls`, then it's just `controls.AfModeEnum...`

Reply:
Done.